### PR TITLE
add section about using `sleep()` from test code to advance time in TS SDK

### PR DESCRIPTION
**Pull request by vkarpov15**
## What does this PR do?

I reviewed https://github.com/temporalio/documentation/pull/1244/files and one neat trick that was missing is the ability to call `testEnv.sleep()` directly from test code to advance time. This was very helpful in https://github.com/vkarpov15/temporal-ecommerce-ts/pull/8, I spent quite a bit of time trying to figure out how to refactor `cartWorkflow()` to make it easier to test with `runUntil()`, before stumbling across [this code](https://github.com/temporalio/sdk-typescript/blob/c25e91309b980f2118df4048d760306982019871/packages/testing/src/index.ts#L285-L287) and realizing that I could skip time from tests.

### CODE REVIEW

File: docs/typescript/testing.md [hunk 1]
```diff
@@ -164,6 +164,54 @@ test('sleep completes almost immediately', async () => {
 });
 ```
 
+The `runUntil()` function runs your Workflow to the end.
```
Comment (bergundy) [inline docs/typescript/testing.md:167]: This is not true, it runs the worker until a promise or async function resolves

File: docs/typescript/testing.md [hunk 2]
```diff
@@ -164,6 +164,54 @@ test('sleep completes almost immediately', async () => {
...
+}
+```
+
+`test.ts`
+
+```ts
+test('advancing time using `testEnv.sleep()`', async function () {
+  const client = testEnv.workflowClient;
+
+  handle = await client.start(sleeperWorkflow, {
```
Comment (bergundy) [inline docs/typescript/testing.md:197]: I'd mention that if we'd have run `execute` or waited for `result` server would transition to skipped time mode, using `start` keeps the server in "normal" time.

**Review by bergundy — APPROVED** [review]
Had a couple minor comments

**Comment by vkarpov15** [comment]
@bergundy thanks for the feedback, fixed :+1:

**Comment by lorensr** [comment]
@temporalio/docs ready for approval

**Review by lorensr — APPROVED** [review]

**Pull request merged**